lux/lib/lux/prisms/discord/messages/pin_message_prism.ex

```diff
@@ -0,0 +1,145 @@
+defmodule Lux.Prisms.Discord.Messages.PinMessagePrism do
+  @moduledoc """
+  A prism for pinning messages in a Discord channel.
+
+  ## Examples
+      iex> PinMessagePrism.handler(%{
+      ...>   channel_id: "123456789",
+      ...>   message_id: "987654321"
+      ...> }, %{agent: %{name: "Agent"}})
+      {:ok, %{pinned: true}}
+  """
+
+  use Lux.Prism,
+    name: "Pin Discord Message",
+    description: "Pins a message in a Discord channel",
+    input_schema: %{
+      type: :object,
+      properties: %{
+        channel_id: %{
+          type: :string,
+          description: "The ID of the channel containing the message",
+          pattern: "^[0-9]{17,20}$"
+        },
+        message_id: %{
+          type: :string,
+          description: "The ID of the message to pin",
+          pattern: "^[0-9]{17,20}$"
+        },
+        reason: %{
```

Comment:
I think that since many of these arguments are optional and perhaps not really needed for most day to day operations, we might want to omit them.